scan-payment/src/main/java/com/getbouncer/scan/payment/ml/TextDetector.kt

```diff
@@ -0,0 +1,370 @@
+                    nameBoxes.box.rect,
+                    nameBoxes.box.confidence,
+                    nameBoxes.box.label + NUM_CLASS // making up a new label for debugging purposes
+                )
+            )
+            allObjects.add(
+                DetectionBox(
+                    panBoxes.box.rect,
+                    panBoxes.box.confidence,
+                    panBoxes.box.label + NUM_CLASS // making up a new label for debugging purposes
+                )
+            )
+        }
+        return Prediction(
+            allObjects,
+            nameBoxes?.subBoxes ?: emptyList(),
+            outputBoxes.filter { it.label == 0 }.maxBy { it.confidence }
+        )
+    }
+
+    /**
+     * Find all boxes that are "close" to the [originalBox]. The returned list of boxes is sorted
+     * from leftmost to rightmost (by the x value)
+     *
+     * A box is defined as close to the [originalBox] if:
+     * 1. it is the [originalBox]
+     * 2. it's not an Expiry box (Expiry boxes should never be merged)
+     * 3. the top of the boxes are mostly aligned
+     * 4. The height of the box is similar to the height of the original box
+     */
+    private fun getCloseBoxes(originalBox: DetectionBox, boxes: List<DetectionBox>): List<DetectionBox> {
+        fun isSimilarYStart(it: DetectionBox) = abs(it.rect.top - originalBox.rect.top) <
+            originalBox.rect.height() * 0.4
+        fun isSimilarHeight(it: DetectionBox) = 0.7 < it.rect.height() / originalBox.rect.height() &&
+            (it.rect.height() / originalBox.rect.height()) < 1.3
+        return boxes.filter {
+            (it == originalBox) || (it.label != 0 && isSimilarHeight(it)) && isSimilarYStart(it)
+        }.sortedBy { it.rect.left }
+    }
+
+    // Run NMS on detection results
+    private fun extractPredictions(raw: List<DetectionBox>): List<DetectionBox> {
+        val predictions = mutableListOf<DetectionBox>()
+
+        val scores = raw.map { it.confidence }
+        val boxes = raw.map { rectForm(it.rect.left, it.rect.top, it.rect.right, it.rect.bottom) }
+
+        val indexes =
+            hardNonMaximumSuppression(
+                boxes = boxes.toTypedArray(),
+                probabilities = scores.toFloatArray(),
+                iouThreshold = 0.4f,
+                limit = null
+            )
+
+        for (index in indexes) {
+            predictions.add(
+                raw[index]
+            )
+        }
+        return predictions
+    }
+
+    /**
+     * From a list of raw boxes, determine the likely PAN location and the name
+     */
+    private fun getNameBox(boxes: List<DetectionBox>): Pair<MergedBox, MergedBox?>? {
+        val metaBoxes = mergeAllBoxes(boxes)
+        val panMetaBox = getPanBox(metaBoxes) ?: return null
+        return Pair(
+            panMetaBox,
+            metaBoxes.asSequence().filter {
+                // filter for merged boxes that are classified as text
+                it.box.label == 1
```

Comment:
can we make the label `1` a constant somewhere so it's clear that it means TEXT?